### PR TITLE
[2.14.0] WS2-2089: New Component: Animated Quote

### PR DESCRIPTION
**Pull request by juanmitriatti**
### Description
New Component: Animated Quote

#### Solution 
Added new plugins with blockquote animated option for ckeditor
ckeditor1
ckeditor2
ckeditor3

### Links

- [JIRA ticket](https://asudev.jira.com/browse/WS2-2089)
- 
Example PAGE : 
https://pr-570-webspark-ci.ws.asu.edu/blockquote-animated
### QA Steps

- Pull the branch locally into your existing env (do this instead of using the PR multidev so you can test the update hooks).
- Run `drush updb`. Update hook `webspark_update_10012` should run without error.
- Create or edit a Basic Page.
- Verify that the "Blockquote Animate" and "Highlight" options now appear on the CKEditor toolbar.
- Ensure the "Highlight" option is disabled by default.
- Add a new "Blockquote Animate."
- When you click on the blockquote content, confirm that the "Highlight" option becomes enabled. Then, add content as usual and verify the animated effect over the this new plugin content.
- In your local env, create a backup of your database as needed.
- Wipe the database, and perform a new installation.
- Create a new Basic page.
- Ensure the new plugin appear by default on basic html and full html option for Ckeditor.

### Checklist

- [ ] Design updates match [Web Standards](https://xd.adobe.com/view/56f6cb78-9af5-4b12-b4ce-ef319f71113f-03a5/) and [Unity Design System](https://unity.web.asu.edu)
- [ ] Solution is documented on the Jira ticket
- [ ] QA steps to verify have been included on the Jira ticket
- [ ] No new PHP or JS errors
- [ ] No accessibility issues are introduced with this update
- [ ] Added/updated README.md files, if relevant
- [ ] Confirm that any yaml files included have a matching update hook

### Verified in browsers 

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Screenshots

Note: Sections that are not applicable for this issue can be removed.

### CODE REVIEW

**Comment by mlsamuelson** [comment]
[Visit Site](https://pr-570-webspark-ci.pantheonsite.io/)

Created multidev environment [pr-570](https://dashboard.pantheon.io/sites/aec2f75f-35eb-41f8-abe7-95b292415259#pr-570) for webspark-ci.

**Review by mlsamuelson — COMMENTED** [review]
Notes from our Zoom discussion:
* Next step is to create a story for the Animated Quote component in the `unity-bootstrap-theme` component in UDS (for UDS-1758) to serve as the reference story for this component,
* Then rework this PR and implement that story as an Animated Quote plugin in CKEditor that has the following features:
  * an icon to add Animated Quote
  * an icon to highlight text. It should be disabled unless the focus is in an Animated Quote. In the future we could have the option to make this usable in other components, but for now let's limit it to this one.
  * the highlight colors should be limited to what's in the mockup.

**Review by mlsamuelson — CHANGES_REQUESTED** [review]
There is a merge conflict to resolve.

Can we please locate the two buttons so they are side-by-side? Can the animated blockquote button be first, and then the highlight button right after it? This will improve the UX.

Also, I've noticed when you make a link in the new quote in the editor, that the link is maroon instead of black.  I think that was an issue some of the other plugins had. When you save and view the rendered output, the link is the correct (black) color. Hopefully that's an easy fix.

File: web/modules/webspark/webspark_ckeditor_plugins/js/ckeditor5_plugins/websparkPlugin/src/animatedtext/css/animatedtext.css
Comment (mlsamuelson) [inline web/modules/webspark/webspark_ckeditor_plugins/js/ckeditor5_plugins/websparkPlugin/src/animatedtext/css/animatedtext.css:1]: Let's redo this file so it only includes the styles we're using, and it also should not include another project's copyright.

File: web/modules/webspark/webspark_ckeditor_plugins/js/ckeditor5_plugins/websparkPlugin/src/animatedtext/websparkanimatedtextui.js [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+  get localizedOptionTitles() {
+    const t = this.editor.t;
+
+    return {
+      "Yellow marker": t("Yellow marker"),
+      "Green marker": t("Green marker"),
+      "Pink marker": t("Pink marker"),
+      "Blue marker": t("Blue marker"),
+      "Red pen": t("Red pen"),
+      "Green pen": t("Green pen"),
```
Comment (mlsamuelson) [inline web/modules/webspark/webspark_ckeditor_plugins/js/ckeditor5_plugins/websparkPlugin/src/animatedtext/websparkanimatedtextui.js:29]: Can we remove the unused options?

File: web/themes/webspark/renovation/src/sass/helpers/_ckeditor.scss
Comment (mlsamuelson) [inline web/themes/webspark/renovation/src/sass/helpers/_ckeditor.scss:1]: We can remove the non-yellow color declarations from this file.

**Review by mlsamuelson — CHANGES_REQUESTED** [review]

**Comment by tbutterf** [comment]
@juanmitriatti I have followed the QA steps, and the plugin buttons do not appear by default on a new install. I have attempted several clean installs, and the buttons are not added by default. They are available to be added by an administrator, but they are not appearing by default. I think maybe you need to import some configs so that they appear by default.

**Comment by tbutterf** [comment]
Also, not everyone is going to want to have a "Title" for their block quote. However, if I leave that field blank, the space reserved for it does not go away. It would be good to have it so that if a field is not used, the space it would have taken up should be removed.

**Comment by tbutterf** [comment]
One last thing. On the clean install I attempted, I went in and added the two buttons as a site admin. Afterwards, when I tried to use it, the highlight did not animate. It looks like this:
image

**Comment by tbutterf** [comment]
There's a problem with tabbed browsing order. When you are in the "Title" field, if you tab to go to the "Content" field, it skips over that field entirely and goes to "Citation name" instead.

**Comment by juanmitriatti** [comment]
All the issues reported were addressed. Please review again.
CC @mlsamuelson  @tbutterf

File: web/themes/webspark/renovation/src/sass/helpers/_ckeditor.scss
Comment (juanmitriatti) [reply web/themes/webspark/renovation/src/sass/helpers/_ckeditor.scss:1]: Fixed, please review again.

**Comment by juanmitriatti** [comment]
> Also, not everyone is going to want to have a "Title" for their block quote. However, if I leave that field blank, the h4 space reserved for it does not go away. It should be that if a field is not used, the space it would have taken up should not be included in the markup. image

Fixed, please review again.

**Comment by mlsamuelson** [comment]
Still seeing the issue with Content textarea not being focusable.
Also there is a regression with the styles not displaying in the Edit and View modes.
view-basic-page
editor-basic-and-full-html

**Comment by mlsamuelson** [comment]
It seems the content textarea isn't focusable in the other plugins as well. And in the Buttons plugin the selection fields aren't focusable. Hopefully that can be fixed for all of these.

**Comment by mlsamuelson** [comment]
My bad. I had not run the `yarn production` command to rebuild the theme so the styles weren't showing.

So that's not an issue.

**Review by mlsamuelson — APPROVED** [review]

**Pull request merged**